clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -579,7 +579,7 @@ public void testConstructorWithNotStringKey() {
         ConfigException ce = assertThrows(
             ConfigException.class,
             () -> new KafkaProducer<>(props, new StringSerializer(), new StringSerializer()));
-        assertTrue(ce.getMessage().contains("not string key"), "Unexpected exception message: " + ce.getMessage());
+        assertTrue(ce.getMessage().contains("not a string"), "Unexpected exception message: " + ce.getMessage());
```

Comment:
You could match on the complete error message (`"One or more keys is not a string."`), right?